controllers/harbor/harbor.go

```diff
@@ -30,18 +30,28 @@ func (harbor *HarborReconciler) Reconcile() (*lcm.CRStatus, error) {
 		if errors.IsNotFound(err) {
 			return harbor.Provision()
 		} else {
-			// TODO given clear reason and message.
-			return harborCRNotReadyStatus("", ""), err
+			return harborClusterCRUnknownStatus(), err
 		}
 	}
-	return nil, nil
-}
+	harbor.CurrentHarborCR = &harborCR
+	isScalingEvent := harbor.isScalingEvent(harbor.HarborCluster, &harborCR)
```

Comment:
`isScalingEvent` and `isUpdatingEvent` are based on the similar parameters. So, maybe we can define a function like 

```go
 // return different reconcile const values
//  "Scaling"
//  "Udating"
  harbor.checkReconcileKind(harbor.HarborCluster, &harborCR)
```

Then here we can use `switch` to organize the actions (Scale()/Update())

Reply:
It can't be handled with `switch` function in one reconciler process when it contains both updates and scale events.